packages/calypso-e2e/src/data-helper.ts

```diff
@@ -34,16 +34,39 @@ export function getCalypsoURL(
 /**
  * Returns the credential for a specified account from the configuration file.
  *
- * @param {string} username Username of the account for which the password is to be obtained.
- * @returns {string} Credential entry found in the configuration file for the given username.
+ * @param {string} username Username of the account for which the credentials are to be obtained.
+ * @returns {string[]} Credential entry found in the configuration file for the given username.
  * @throws {Error} If username does not correspond to a valid entry in the configuration file.
  */
-export function getAccountCredential( username: string ): string {
+export function getAccountCredential( username: string ): string[] {
 	const testAccounts: { [ key: string ]: string } = config.get( 'testAccounts' );
 	try {
-		return testAccounts[ username ];
+		const [ user_name, password ] = testAccounts[ username ];
+		return [ user_name, password ];
 	} catch ( err ) {
-		throw new Error( `Credential for username: ${ username } not found in configuration file.` );
+		throw new Error(
+			`Configuration file did not contain credentials for requested user ${ username }.`
+		);
+	}
+}
+
+/**
+ * Returns the site URL for a specified account from the configuration file.
+ *
+ * @param {string} username Username of the account for which the site URL is to be obtained.
+ * @returns {string} Site URL for the given username.
+ * @throws {Error} If the username does not have a corresponding site URL, or username does not have an entry in the configuration file.
+ */
+export function getAccountSiteURL( username: string ): string {
+	const testAccounts: { [ key: string ]: string } = config.get( 'testAccounts' );
+	try {
+		const [ , , url ] = testAccounts[ username ];
+		if ( url ) {
+			return `https://${ url }`;
+		}
+		throw new Error();
```

Comment:
Unless related to https://github.com/Automattic/wp-calypso/pull/53310#discussion_r648139543

Reply:
Interesting....how did this make it here?

Reply:
Oh, I remember now.

If `throw new Error()` at Line 67 is not there, then TypeScript transpiler complains about not all paths return a value.

The `throw` there is to ensure that if `url` was undefined, an error is thrown which is caught in the next line.

Reply:
What do you think about the function throwing an error as opposed to say, returning an empty string if `url` is not defined in the secrets file? 

Personally I like the approach of throwing since it would halt execution immediately, since not having a valid `url` to visit is not a problem that the framework should correct.